lib/sqs.js

```diff
@@ -4,6 +4,27 @@ const _ = require('lodash');
 const AWS = require('aws-sdk');
 const Promise = require('bluebird');
 const clientConfig = require('./client-config');
+const uuid = require('uuid/v4');
+const util = require('util');
+const zlib = require('zlib');
+const compress = util.promisify(zlib.gzip);
+const decompress = util.promisify(zlib.gunzip);
+
+// 256kb => b
+const MAX_MSG_SIZE = 262144;
+
+/**
+ * Compute the size of an SQS message, which is -- the sum of the base64-encoded
+ * message body + the size of message attribute keys + values.
+ * 
+ * @param {*} body an SQS message body
+ * @param {*} attrs a map of SQS message attributes
+ */
+function computeMessageSize(body, attrs) {
+  const attrsSize = _.reduce(attrs, (sum, value, key) => sum + Buffer.byteLength(value) + Buffer.byteLength(key)) || 0;
```

Comment:
don't we also have to check byteLength in base64 here?

Reply:
we don't, because the attributes are not sent as base64, only the body